uefi/src/proto/scsi/mod.rs

```diff
@@ -0,0 +1,376 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! SCSI Bus specific protocols.
+
+use crate::helpers::AlignedBuffer;
+use core::alloc::LayoutError;
+use core::error::Error;
+use core::ffi::c_void;
+use core::marker::PhantomData;
+use core::time::Duration;
+use core::{fmt, ptr};
+use uefi_raw::protocol::scsi::{
+    ScsiIoDataDirection, ScsiIoHostAdapterStatus, ScsiIoScsiRequestPacket, ScsiIoTargetStatus,
+};
+
+pub mod pass_thru;
+
+/// Represents the data direction for a SCSI request.
+///
+/// Used to specify whether the request involves reading, writing, or bidirectional data transfer.
+pub type ScsiRequestDirection = uefi_raw::protocol::scsi::ScsiIoDataDirection;
+
+/// The `AlignmentError` is returned if a user-provided buffer doesn't fulfill alignment requirements.
+///
+/// This ensures that all buffers used in SCSI commands conform to the alignment requirements of the protocol.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct AlignmentError;
```

Comment:
this struct and change belongs to the previous commit introducing the AlignedBuffer struct I think